File: care_well_backend/CareWell/CareWell.Domain/Salud/FichaSaludAlergia.cs
using CareWell.Domain.ValueObjects.Salud;
using CareWell.Global.Exceptions;
using CareWell.Global.Mensajes;

namespace CareWell.Domain.Salud
{
    public class FichaSaludAlergia : BaseEntity
    {
        public virtual FichaSalud FichaSalud { get; private set; }

        public virtual string Nombre { get; private set; }

        public virtual string Reaccion { get; private set; }

        public virtual string? Medicamento { get; private set; }

        public virtual void Registrar(AgregarAlergia agregarAlergia, FichaSalud fichaSalud)
        {
            if (string.IsNullOrEmpty(agregarAlergia.Nombre))
                throw new ValidacionDominioException(Mensajes.NombreAlergiaRequerido);

            if (string.IsNullOrEmpty(agregarAlergia.Reaccion))
                throw new ValidacionDominioException(Mensajes.ReaccionAlergiaRequerido);

            this.FichaSalud = fichaSalud;
            this.Nombre = agregarAlergia.Nombre;
            this.Reaccion = agregarAlergia.Reaccion;
            this.Medicamento = agregarAlergia.Medicamento;
        }
    }
}

File: care_well_app/lib/domain/entities/health/ficha_salud_alergia.dart
import 'package:care_well_app/domain/entities/entities.dart';

class FichaSaludAlergia extends BaseEntity {
  final String nombre;
  final String reaccion;
  final String? medicamento;

  const FichaSaludAlergia({
    super.id = 0,
    required this.nombre,
    required this.reaccion,
    this.medicamento,
  });

  @override
  FichaSaludAlergia copyWith({
    int? id,
    String? nombre,
    String? reaccion,
    String? medicamento,
  }) {
    return FichaSaludAlergia(
      id: id ?? this.id,
      nombre: nombre ?? this.nombre,
      reaccion: reaccion ?? this.reaccion,
      medicamento: medicamento,
    );
  }
}
